tests/qnodes/test_qnode_base.py

```diff
@@ -781,6 +781,39 @@ def circuit(w, x=None, **kwargs):
         c = node(1.0, x=np.pi, y=10)
         assert c == pytest.approx(-1.0, abs=tol)
 
+    def test_mutable_node_variable_number_of_args(self, qubit_device_1_wire):
+        """Tests a mutable circuit that expects a variable number of
+        positional arguments based on the values of its auxiliary arguments.
+        """
+        def circuit(x, n=1):
+            for k in range(n):
+                qml.RX(x[k], wires=0)
+            return qml.expval(qml.PauliZ(wires=0))
+
+        node = BaseQNode(circuit, qubit_device_1_wire, mutable=True)
+
+        # first evaluation creates VariableRefs for the positional args
+        node([0.1], n=1)
+        # new VariableRefs must be created since the auxiliary args have changed
+        node([0.1, 0.2], n=2)
```

Comment:
It would be nice to check the values against the analytical values as well. The reason would be that the test should not only check the mutable QNode case but correctness as well.

Reply:
Maybe not correctness (since it might verge on an integration test), but it is worth checking that the qnode has the correct value for number of args after each call.

Reply:
Added.